src/data-models/mdm-data-model.resource.ts

```diff
@@ -535,4 +536,22 @@ export class MdmDataModelResource extends MdmModelDomainResource {
     const url = `${this.apiEndpoint}/dataModels/${sourceId}/intersectsMany`;
     return this.simplePost(url, data, options);
   }
+
+  /**
+   * `HTTP PUT` - Copy a Data Model to another Data Model. Define the target folder and the name of the new Data Model.
+   *
+   * @param dataModelId The unique identifier of the Data Model to copy.
+   * @param CopyContainerPayload The payload containing details on where to copy the Data Model to.
+   * @returns The result of the `PUT` request.
+   *
+   * `200 OK` - will return a {@link DataModelDetailResponse} containing a {@link DataModelDetail
+   */
+  copy(
+    dataModelId: Uuid,
+    copyContainerPayload: CopyContainerPayload
```

Comment:
More consistency things:

- [ ] Move `target...` and `source...` properties into function parameters
- [ ] `copyContainerPayload` is not the correct name for this, this implies we're copying a container (like a folder) whereas this is for a model. `CopyModelPayload` would make more sense

Reply:
target folder is part of the payload on this one